File: Python-For-Everyone-Horstmann/Chapter4-Loops/P4.21.py
# Write a program that reads an integer and displays, using asterisks, a filled and hol-
# low square, placed next to each other. For example if the side length is 5, the pro gram
# should display
#     *****   *****
#     *****   *   *
#     *****   *   *
#     *****   *   *
#     *****   *****

side = int(input("Enter side length: "))

lineNum =  1
dotNum = 1

for i in range(side):
    dotNum = 1

    # filled square
    for i in range(side):
        print("*", end = "")

    print(end=" ")

    # hollow square
    for i in range(side):
        if lineNum == 1 or lineNum == side:
            print("*", end = "")

        else:
            if dotNum == 1 or dotNum == side:
                print("*", end = "")

            else:
                print(" ", end = "")

        dotNum += 1

    lineNum += 1
    print()

File: Java-for-Everyone-Late-Objects-Horstmann/Chapter-4-Loops/P4_21.java
/*Write a program that reads an integer and displays, using asterisks, a filled diamond
 of the given side length. For example, if the side length is 4, the program should
 display
     *
    ***
   *****
  *******
   *****
    ***
     * */

import java.util.Scanner;

public class P4_21 {
    public static void main(String[] args) {
	Scanner input = new Scanner(System.in);
	System.out.print("Side length: ");
	int side_length = input.nextInt();
	input.close();
	int max_width = side_length * 2 - 1;
	int half_width = max_width / 2;
	int dot_num = 1;
	// top half
	for (int i = 0; i < side_length; i++) {
	    for (int j = 0; j < half_width; j++) {
		System.out.print(" ");
	    }
	    for (int j = 0; j < dot_num; j++) {
		System.out.print("*");
	    }
	    System.out.println();
	    dot_num += 2;
	    half_width -= 1;
	}
	// bottom half
	dot_num = max_width - 2;
	half_width = 1;
	for (int i = 0; i < side_length; i++) {
	    for (int j = 0; j < half_width; j++) {
		System.out.print(" ");
	    }
	    for (int j = 0; j < dot_num; j++) {
		System.out.print("*");
	    }
	    System.out.println();
	    dot_num -= 2;
	    half_width += 1;
	}
    }
}
